config_parser_test.cc

```diff
@@ -1,11 +1,68 @@
 #include "gtest/gtest.h"
 #include "config_parser.h"
 
-TEST(NginxConfigParserTest, SimpleConfig) {
-  NginxConfigParser parser;
-  NginxConfig out_config;
+// foo bar;
+TEST(NginxConfigTest, ToString) {
+    NginxConfigStatement statement;
+    statement.tokens_.push_back("foo");
+    statement.tokens_.push_back("bar");
+    EXPECT_EQ(statement.ToString(0), "foo bar;\n");
+}
+
+class NginxStringConfigTest : public ::testing::Test {
+	protected:
+    bool ParseString(const std::string config_string) {
+        std::stringstream config_stream(config_string);
+        return parser_.Parse(&config_stream, &out_config_);
+    }
+    NginxConfigParser parser_;
+    NginxConfig out_config_;
+};
 
-  bool success = parser.Parse("example_config", &out_config);
+TEST_F(NginxStringConfigTest, SyntaxErrorConfig) {
```

Comment:
Can separate some of these examples from class into their own tests so that they are grouped in a way that's easier to understand. For ex. the tests for NestedConfig, CommentsConfig, and QuotesConfig are all types of SyntaxError tests and contain some duplicates with tests in SyntaxErrorConfig.